### ISSUE TITLE
Plans for the second version

### ISSUE THREAD

**Issue by fejes713** [30-seconds/30-seconds-of-interviews#171]
Hello everyone 👋 

It's been over a year since this project was released. The feedback so far was phenomenal and it' seems that some people really find this project useful despite my initial thoughts. 

I'd like to make the second version of it and I need your opinions on it. The idea is to introduce a completely new structure similar to the 30-seconds-starter kit. 

Things I would like to include in the second version: 
- Make the project 100% accessible, and add accessibility category
- Introduce visually pleasing dark mode
- At least one new category (e.g. Computer Science)
- Make it with Gatsby (so many things out of the box)
- Add the ability to share questions
- Individual pages for each question (better SEO I guess)
- Remove expertise tags, at least visually (so many people complain about it)
- Add about page
- A form to submit new questions without GitHub account
- Host the project on Netlify
- A random interview simulation

What do you think about proposed changes? Would you like to see something else? What current parts of the projects do you like the most?

**Comment by jeremydouglass** [comment]
-  Make the project 100% accessible, and add accessibility category

This is a really excellent idea!

-  A random interview simulation

Interesting, although maybe not as high priority? If it would be a random or sequenced collection of n questions from selected subsections of these lists, it might be important to be clear that it would simulate only one part of an interview -- not all the other kinds of questions about design problems and soft skills et cetera.

**Comment by AlejandroMataGuillen** [comment]
Multiple languages? maybe some people can help with the translation. I can help with Spanish for example

**Comment by Chalarangelo** [comment]
**The features I would prioritize for a v2 are as follows:**

> - Make the project 100% accessible, and add accessibility category
> - Introduce visually pleasing dark mode
> - Make it with Gatsby (so many things out of the box)
> - Individual pages for each question (better SEO I guess)
> - Host the project on Netlify

**About the leftover points:**
> - At least one new category (e.g. Computer Science)

Content should not be part of a UI redesign, if you want the launch to coincide with the new content, it's entirely ok, but let's keep them separate as to not lose track.

> - Add the ability to share questions

Social media sharing is something of a complicated bit, as I realized with the 30-seconds-starter. Lower priority to this also, due to the fact that individual questions will have their own page and are easier to share as links.


> - Remove expertise tags, at least visually (so many people complain about it)

I am honestly very indifferent about this. If people don't like it, just hide the expertise tag or remove altogether during the migration phase. It's not up to me to decide priorities on this one at all, let the people speak!


> - Add about page

Nice to have, can be done later (e.g. right after launch). If there are many moving parts on the redesign and new version, it can actually be a placeholder for a day or two, Google will take forever to index everything anyways.


> - A form to submit new questions without GitHub account

That might be a bit of a hassle in terms of managing afterwards. I suspect these will end up in a Google Forms or an email or similar. Do we want to commit to working on those? GitHub takes 5 minutes to set up and we can provide feedback, have a discussion, help people out to make a contribution and get their name on the Contributors list. I would vote against this idea.

> - A random interview simulation

Could be interesting, but it might not be that important a feature. We need to see use-cases and figure out if people want to study for an interview or simulate one. I guess the former, but it remains to be seen.

> - Multiple languages? maybe some people can help with the translation. I can help with Spanish for example

The reality is that multilingual has always run into issues with translators and commitment, plus the difficulty of translating certain terms in some languages. I think if translations are going to happen, we need dedicated team members for these. Plus it's a bit harder to check the quality of explanations in a language we do not understand. I'm not against it, I'm just bringing everyone up to date on the issues we have faced before.

**My 2 cents:**

- Definitely add Analytics again (non-intrusive, GDPR compliant, no tracking)
- Add a sitemap
- Create a proper search system, as a complementary tool to filters

**Comment by atomiks** [comment]
I don't know how this could be included on the site, but a significant part of interviews is "soft skills" and how they the interviewee fits in with the culture of a company or presents themselves, everything we have here is purely hard skills. It's probably hard to adapt this into snippet form but just something I thought about

**Comment by Chalarangelo** [comment]
@atomika great idea. I think that we could have a snippet-less guide or list of soft skills tips for interviews as a separate page on the website. That could do the trick, what do you think?

**Comment by Tikam02** [comment]
@fejes713 Can I contribute to this project by implementing features and also by adding the question.

**Comment by fejes713** [comment]
Hey, @Tikam02 you are welcome to contribute to this repository! Let me know if you need any help with it 😄 I'll be more than happy to help. 

You can start by adding a question that you mentioned. You can find all guides inside the contributing file on this repository. 

Speaking of new version, I am planning to start it late September so if you're available you can definitely help with few things :octocat:

**Comment by skatcat31** [comment]
@fejes713 accessibility is a huge undertaking. At work we use Axe, in either the command line tool or the browser extension to check to make sure we're accessible.

Another section that might be REALLY useful, especially for full stack developers, is proper test writing. An example:

If you are coding for a website, how would you write your test for your component? What about for the page it is going to be rendered in?

If you're writing a test for a back end system, how would you test that functions are getting called correctly by your code? What about external integrations?

These are both HUGE considerations(and are make or break in MANY interviews) so we may want to visit those ideas as well.

**Comment by fejes713** [comment]
Amazing! Thanks for the feedback @skatcat31, **a testing category is a must-have for v2**. I'll think about the questions and consult with my senior peers to review them before we start adding and answering them. Couldn't agree more with you 👍

**Comment by jordanch** [comment]
I came here because I wanted to suggest search, in some form, but see someone has already. Great. 

As an end-user, I'd appreciate being able to search questions by (at least) input string. E.g. If I input "portal" into a search field, filter currently selected category questions by the input, searching across perhaps the question title and body.

The search design should be fleshed out further but herewith emphasis on it's need :)

**Comment by fejes713** [comment]
Hi Jordan 👋 

Thanks for submitting your feedback. I am on my trip home and hope to improve this projects by the end of this year - then we'll definitely include the search and better filtering.
